MainRepository/Program.cs

```diff
@@ -1,7 +1,111 @@
-﻿internal class Program
-{
-    private static void Main(string[] args)
-    {
-        Console.WriteLine("Hello, World!");
-    }
-}
\ No newline at end of file
+﻿namespace List;
+internal class Program {
+   private static void Main () {
+   }
+}
+
+#region class List ---------------------------------------------------------------------
+/// <summary>Implementation of list class</summary>
+/// <typeparam name="T">T is type of variables declared in the List</typeparam>
+public class MyList<T> {
+
+   #region Constructor ------------------------------------------- 
+   /// <summary>Constructor used to assign values to the list and count</summary>
+   public MyList () {
+      mList = new T[4];
+      mCount = 0;
+   }
+   #endregion
+
+   #region Properties ---------------------------------------------
+   /// <summary>Gets the total number of elements in the list</summary>
+   public int Capacity => mList.Length;
+
+   /// <summary>Gets the number of elements in the list</summary>
+   public int Count => mCount;
+   #endregion
+
+   #region Methods
+   /// <summary>Adds an element to the list and increase its size if required</summary>
+   /// <param name="a">The element to be added</param>
+   public void Add (T a) {
+      if (mCount == Capacity) {
+         var tmp = new T[mCount * 2];
+         for (int i = 0; i < mCount; i++) tmp[i] = mList[i];
+         mList = tmp;
+      }
+      mList[mCount++] = a;
+   }
+
+   /// <summary>Clear all elements from the list</summary>
+   public void Clear () {
+      Array.Clear (mList);
+      mCount = 0;
+   }
+
+   /// <summary>Inserts an element at the given index</summary>
+   /// <param name="index">The index position from which the element is to be inserted</param>
+   /// <param name="a">The element to be inserted</param>
+   /// <exception cref="IndexOutOfRangeException">Throws exception when the given index below 0 (or) greater than or equal to the count</exception>
+   public void Insert (int index, T a) {
+      ValidateIndex (index);
+      if (mCount == Capacity) {
+         var tmp = new T[mCount * 2];

```

Comment:
https://github.com/karthick-1205/Assignments/pull/41/files#r1393921307
This is not done